snuba/snuba_migrations/querylog/0006_sorting_key_change.py

```diff
@@ -0,0 +1,156 @@
+    new_sampling_key = "request_id TTL timestamp + toIntervalDay(30)"
+
+    ((curr_sampling_key, curr_sorting_key),) = clickhouse.execute(
+        f"SELECT sampling_key, sorting_key FROM system.tables WHERE name = '{TABLE_NAME}' AND database = '{database}'"
+    ).results
+
+    new_create_table_statement = curr_create_table_statement.replace(
+        TABLE_NAME, TABLE_NAME_NEW
+    )
+
+    # Switch the sorting key
+    if curr_sorting_key != new_sorting_key:
+        assert new_create_table_statement.count(curr_sorting_key) == 1
+        new_create_table_statement = new_create_table_statement.replace(
+            curr_sorting_key, new_sorting_key
+        )
+
+    # Switch the sample by
+    if curr_sampling_key != new_sampling_key:
+        assert new_create_table_statement.count("SAMPLE BY " + curr_sampling_key) == 1
+        new_create_table_statement = new_create_table_statement.replace(
+            "SAMPLE BY " + curr_sampling_key, "SAMPLE BY " + new_sampling_key
+        )
+
+    # Update the timestamp column
+    # Clickhouse 20 does not support altering a column in the primary key so we need to do it here
+    new_timestamp_type = "`timestamp` DateTime CODEC(T64, ZSTD(1))"
+    assert new_create_table_statement.count("`timestamp` DateTime") == 1
+    new_create_table_statement = new_create_table_statement.replace(
+        "`timestamp` DateTime", new_timestamp_type
+    )
+    assert new_timestamp_type in new_create_table_statement
+
+    # Create the new table
+    clickhouse.execute(new_create_table_statement)
+
+    # Copy the data over
+    [(row_count,)] = clickhouse.execute(f"SELECT count() FROM {TABLE_NAME}").results
+    batch_size = 100000
+    batch_count = math.ceil(row_count / batch_size)
+
+    orderby = "toStartOfDay(finish_ts), project_id, event_id"
+
+    for i in range(batch_count):
+        skip = batch_size * i
+        clickhouse.execute(
+            f"""
+            INSERT INTO {TABLE_NAME_NEW}
+            SELECT * FROM {TABLE_NAME}
+            ORDER BY {orderby}
+            LIMIT {batch_size}
+            OFFSET {skip};
+            """
+        )
+
+    clickhouse.execute(f"RENAME TABLE {TABLE_NAME} TO {TABLE_NAME_OLD};")
+
+    clickhouse.execute(f"RENAME TABLE {TABLE_NAME_NEW} TO {TABLE_NAME};")
+
+    # Ensure each table has the same number of rows before deleting the old one
+    assert (
+        clickhouse.execute(f"SELECT COUNT() FROM {TABLE_NAME};").results
+        == clickhouse.execute(f"SELECT COUNT() FROM {TABLE_NAME_OLD};").results
+    )
+
+    clickhouse.execute(f"DROP TABLE {TABLE_NAME_OLD};")
+    return
+
+
+def forwards(logger: logging.Logger) -> None:
+    """
+    This migration is a bit complicated because we need to change the sorting key.
+    We can't do it inplace so we need to create a new table with the modified keys,
+    copy the data over, then drop the old table.
+    """
+    cluster = get_cluster(StorageSetKey.TRANSACTIONS)
```

Comment:
Why is the cluster fetched with `StorageSetKey.TRANSACTIONS`? Is it a typo?

Reply:
fixed